### PR TITLE
Codec Adapter: Add support for multiple input/output buffers - Part 1

### PR DESCRIPTION
**Pull request by ranj063**
Add support for processing samples from multiple input buffers and copying the output to multiple output buffers.

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
How is the routing between input and output done ? Is the mapping stored within the module or within the core pipeline or both ?

**Comment by ranj063** [comment]
> How is the routing between input and output done ? Is the mapping stored within the module or within the core pipeline or both ?

The routing is assumed to be stored within the module and the core pipeline. The adapter code copies blindly today.

**Comment by lgirdwood** [comment]
@ranj063 looks Jenkins CI got blocked, but there is a failure on PR_CODEC so I wont rerun CI until you've had a chance to look at it.

**Comment by ranj063** [comment]
> @ranj063 looks Jenkins CI got blocked, but there is a failure on PR_CODEC so I wont rerun CI until you've had a chance to look at it.

@lgirdwood all fixed now

File: src/audio/codec_adapter/codec_adapter.c [hunk 1]
```diff
@@ -477,6 +517,19 @@ int codec_adapter_copy(struct comp_dev *dev)
...
+	list_for_item(blist, &mod->sink_buffer_list) {
+		struct comp_buffer *buffer = container_of(blist, struct comp_buffer, sink_list);
+
+		if (mod->output_buffers[i].size > 0) {
+			ca_copy_from_module_to_sink(&buffer->stream, mod->output_buffers[i].data,
+						    mod->output_buffers[i].size);
+			audio_stream_produce(&buffer->stream, mod->output_buffers[i].size);
+		}
+		i++;
+	}
```
Comment (lyakh) [inline src/audio/codec_adapter/codec_adapter.c:588]: I find this commit a bit confusing - it adds a list of sink buffers but also keeps the current single sink buffer, right? Below there's a line:
```
audio_stream_produce(&local_buff->stream, md->mpd.produced);
```
Maybe that's removed in a later commit, let's see

**Review by lyakh — COMMENTED** [review]
I didn't find an explanation why multiple output buffers were added without removing the current presumably single output buffer. Probably it's just me not understanding that

**Comment by ranj063** [comment]
@lgirdwood @dbaluta closing this one. Will open 2 separate PRs

**Comment by dbaluta** [comment]
@ranj063 thanks!

**Pull request closed without merging**